src/RequestManager.ts

```diff
@@ -49,12 +52,23 @@ class RequestManager {
   }
   private onData(data: string): void {
     const parsedData = JSON.parse(data);
-    if (typeof parsedData.result === "undefined") {
+    if (typeof parsedData.result === "undefined" && typeof parsedData.error === "undefined") {
       return;
     }
+    const req = this.requests[parsedData.id];
     // call request callback for id
-    if (this.requests[parsedData.id]) {
-      this.requests[parsedData.id](parsedData);
+    if (req) {
```

Comment:
nit. maybe unnest this request

Reply:
```suggestion
    if(req === undefined) return;
```

Reply:
or throw? if this ever happens something is wrong.... no?

Reply:
maybe, maybe not, the problem with that is 3rd party systems . I.e. my rpc service sometimes returns nothing or fires blanks

Reply:
There is a real problem of the transport failing , i.e. what happens when you socket connection gets dropped, the transport will throw , and I assume that bubbles up to the request manager. Does that result in a parsedData.error of undefined?